docs/artifact-descriptions/vmcai26.md

```diff
@@ -0,0 +1,213 @@
+
+Refers only to numbers in Fig. 8.
+
+
+> With these settings, the system ran out of memory for 4 tasks, 3 tasks timed out, and the analyzer crashed for 7 tasks.
+
+The numbers in the VM are essentially the same as those on the server as Goblint is inherently single-threaded, and each of the cores of the server is not too powerful. Our run inside the VM matched exactly these numbers, however, some wiggle between machines is to be expected.
+
+The numbers can be inspected by clicking on the `Table` tab on top of the document.
+Then, change the dropdown for one of the configs from `Show all` to:
+
+- Select `OUT OF MEMORY` to see how often Goblint ran out of memory
+- Select `TIMEOUT` to see how often Goblint ran into a timeout
+- Select `EXCEPTION(*)` to see the cases where Goblint crashed.
+
+> Where execution terminated, it did so within at most 20s, where for all but 6 tasks the runtime was below 10s.
+
+The same caveat as above applies here too.
+The numbers can be inspected by clicking on the `Table` tab on top of the document.
+Then, sort the runtime in a column by wall time by clicking on `walltime` twice.
+In our run inside the artifact, there in fact were no tasks which ran more than `9s`, as each core in our machine for the VM is a bit more powerful than the cores in the server. Your run likely will not yield identical numbers, but they will be in the same ballpark.
+
+
+
+The *Table* tab also gives access to detailed evaluation results for each file.
+For this to work, first unpack the `*.logfiles.zip` folder located in the `results` folder.
+Clicking on a status then shows the complete log for the benchmark run.
+
+> **Note:** If you are trying to view logs for individual runs through the HTML table (by clicking on the evaluation result `true` or `false`), you may encounter a warning because browsers block access to local files. Follow the instructions in the message to enable log viewing.
+
+
+## Source Code
+
+The Goblint analyzer (<https://goblint.in.tum.de>) is developed by Technical University of Munich and University of Tartu with one of the authors recently having moved from TUM to National University of Singapore. The source code for Goblint at the time of evaluation can be found in this artifact in the `~/analyzer` directory.
+
+For an explanation of the general structure of the system, please refer to the online documentation of the system: https://goblint.readthedocs.io/en/latest/
+
+The type of code particularly relevant for this paper is best illustrated along an example:
+  - A digest (lightweight thread ids in Fig. 4)
+    - Consider the file `~/analyzer/src/analyses/threadFlag.ml` which implements a slightly more powerful version of the lightweight thread ids in Fig. 4
+    - $\cal A$ is defined in `~/analyzer/src/cdomains/threadFlagDomain.ml` (Simple)
+    - The functions `threadenter` and `threadspawn` take care of computing appropriate successors
+    - ||^? is implemented in the function `may_race`
+      ~~~ocaml
```

Comment:
This code block also doesn't render correctly: https://goblint--1893.org.readthedocs.build/en/1893/artifact-descriptions/vmcai26/.